libs/components/forms/src/lib/modules/file-attachment/file-drop/file-drop.component.ts

```diff
@@ -254,22 +263,84 @@ export class SkyFileDropComponent implements OnDestroy {
 
   #_minFileSize = MIN_FILE_SIZE_DEFAULT;
 
+  #notifyTouched: (() => void) | undefined;
+  #notifyChange:
+    | ((_: (SkyFileItem | SkyFileLink)[] | undefined | null) => void)
+    | undefined;
+  #_uploadedFiles: (SkyFileItem | SkyFileLink)[] = [];
+
   readonly #fileAttachmentService = inject(SkyFileAttachmentService);
+  readonly #fileReaderSvc = inject(SkyFileReaderService);
   readonly #liveAnnouncerSvc = inject(SkyLiveAnnouncerService);
   readonly #resourcesSvc = inject(SkyLibResourcesService);
   readonly #idSvc = inject(SkyIdService);
 
   protected errorId = this.#idSvc.generateId();
+
+  protected ngControl = inject(NgControl, { optional: true });
+
   protected rejectedFiles: SkyFileItem[] = [];
 
+  constructor() {
+    if (this.ngControl) {
+      this.ngControl.valueAccessor = this;
+    }
+  }
+
   public ngOnDestroy(): void {
     this.filesChanged.complete();
     this.linkChanged.complete();
     this.linkInputBlur.complete();
   }
 
+  public writeValue(value: any): void {
+    if (value instanceof Array) {
+      const linkUploads: SkyFileLink[] = [];
+      const fileUploads: SkyFileItem[] = [];
+
+      value.forEach((file) => {
+        if ('url' in file && file.url !== undefined) {
+          if (!('file' in file)) {
+            linkUploads.push(file);
+          } else if ('file' in file && file.file !== undefined) {
+            fileUploads.push(file);
+          }
+        }
+      });
+
+      if (!(linkUploads.length > 0) && !(fileUploads.length > 0)) {
+        this.#notifyChange?.(null);
+      } else {
+        this.#_uploadedFiles = [];
+
+        if (linkUploads.length > 0) {
+          linkUploads.forEach((file) => {
+            this.uploadLink(file);
+          });
+        }
+        if (fileUploads.length > 0) {
+          // this prevents FormControl from setting an invalid value before the async
+          // processes in #handleFile is complete
+          this.#notifyChange?.(null);
```

Comment:
I removed this line and the unit tests still pass. Is it possible to write a scenario in the spec that proves your comment's assertion?